File: app/src/main/java/nithra/gift/suggestion/shop/birthday/marriage/Otp/OtpVerify.java
package nithra.gift.suggestion.shop.birthday.marriage.Otp;

import android.content.Intent;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.google.gson.Gson;

import java.util.ArrayList;
import java.util.HashMap;

import nithra.gift.suggestion.shop.birthday.marriage.MyProduct;
import nithra.gift.suggestion.shop.birthday.marriage.R;
import nithra.gift.suggestion.shop.birthday.marriage.Retrofit.CheckOtp;
import nithra.gift.suggestion.shop.birthday.marriage.Retrofit.RetrofitAPI;
import nithra.gift.suggestion.shop.birthday.marriage.Retrofit.RetrofitApiClient;
import nithra.gift.suggestion.shop.birthday.marriage.Retrofit.SendOtppojo;
import nithra.gift.suggestion.shop.birthday.marriage.SellerProfileProductList;
import nithra.gift.suggestion.shop.birthday.marriage.SharedPreference;
import nithra.gift.suggestion.shop.birthday.marriage.Utils_Class;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class OtpVerify extends AppCompatActivity {
    TextView enterotp,edit_email;
    EditText enter_otp, otp_1, otp_2, otp_3, otp_4;
    String verify, edit_otp;
    SharedPreference sharedPreference = new SharedPreference();
    ArrayList<SendOtppojo> send_otp;
    ImageView edit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        this.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
        setContentView(R.layout.fragment_enter_o_t_p);

        TextView _tv = findViewById(R.id.timer);
        send_otp = new ArrayList<SendOtppojo>();

        enterotp = findViewById(R.id.enterotp);
        edit_email = findViewById(R.id.edit_email);
        otp_1 = findViewById(R.id.otp_1);
        otp_2 = findViewById(R.id.otp_2);
        otp_3 = findViewById(R.id.otp_3);
        otp_4 = findViewById(R.id.otp_4);
        edit = findViewById(R.id.edit);

        enter_otp = findViewById(R.id.enter_otp);
        edit_email.setText(sharedPreference.getString(OtpVerify.this, "user_mail"));
        new CountDownTimer(120000, 1000) { // adjust the milli seconds here
            public void onTick(long millisUntilFinished) {
                _tv.setText("If you didn't receive a otp? " + millisUntilFinished / 1000);
            }

            public void onFinish() {
                _tv.setText("If you didn't receive a otp? Resend");
                sharedPreference.putString(OtpVerify.this, "register_otp", "" + 0);
            }
        }.start();


        _tv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //sharedPreference.putString(OtpVerify.this, "register_otp_1", "register_otp");
                otp_generate();
                new CountDownTimer(120000, 1000) { // adjust the milli seconds here

                    public void onTick(long millisUntilFinished) {

                        _tv.setText("If you didn't receive a otp? " + millisUntilFinished / 1000);
                    }

                    public void onFinish() {
                        _tv.setText("If you didn't receive a otp? Resend");
                    }

                }.start();

            }
        });

        edit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
               finish();
            }
        });

        otp_1.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                otp_2.requestFocus();
            }
        });

        otp_2.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                otp_3.requestFocus();

            }
        });

        otp_3.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                otp_4.requestFocus();
            }
        });

        enterotp.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                edit_otp = otp_1.getText().toString().trim() + otp_2.getText().toString().trim() + otp_3.getText().toString().trim() + otp_4.getText().toString().trim();
                if (otp_1.getText().toString().equals("") && otp_2.getText().toString().equals("") && otp_3.getText().toString().equals("") && otp_4.getText().toString().equals("")) {
                    Utils_Class.toast_center(OtpVerify.this, "Enter your otp");
                } else if (sharedPreference.getString(OtpVerify.this, "register_otp").equals(edit_otp)) {
                    System.out.println("printop" + edit_otp);
                    otp_verify();
                } else {
                    Utils_Class.toast_center(OtpVerify.this, "Invalid otp");
                }
            }
        });
    }

    public void otp_verify() {
        verify = edit_otp.trim();

        HashMap<String, String> map = new HashMap<>();
        map.put("action", "check_otp");
        map.put("user_id", sharedPreference.getString(OtpVerify.this, "user_id"));
        //map.put("mobile_num", "" + sharedPreference.getString(OtpVerify.this, "resend"));
        map.put("otp", verify);
        RetrofitAPI retrofitAPI = RetrofitApiClient.getRetrofit().create(RetrofitAPI.class);
        Call<ArrayList<CheckOtp>> call = retrofitAPI.checkotp(map);
        call.enqueue(new Callback<ArrayList<CheckOtp>>() {
            @Override
            public void onResponse(Call<ArrayList<CheckOtp>> call, Response<ArrayList<CheckOtp>> response) {
                if (response.isSuccessful()) {
                    String result = new Gson().toJson(response.body());
                    System.out.println("======response result:" + result);
                    sharedPreference.putInt(OtpVerify.this, "yes", 1);
                    sharedPreference.putInt(OtpVerify.this, "profile", 1);
                    enter_otp.getText().clear();
                    String user = sharedPreference.getString(OtpVerify.this, "user_status");

                    if (user.equals("exiting")) {
                        sharedPreference.putInt(OtpVerify.this, "profile", 2);
                        Intent i = new Intent(OtpVerify.this, SellerProfileProductList.class);
                        startActivity(i);
                        finish();
                        //fragMove.product();
                        //fragMove.seller();
                    } else if (user.equals("new")) {
                        Intent i = new Intent(OtpVerify.this, ShopAdd.class);
                        startActivity(i);
                        finish();
                        //fragMove.seller();

                    }

                }
                System.out.println("======response :" + response);
            }

            @Override
            public void onFailure(Call<ArrayList<CheckOtp>> call, Throwable t) {
                System.out.println("======response t:" + t);
            }
        });
    }


    public void otp_generate() {
        HashMap<String, String> map = new HashMap<>();
        map.put("action", "check_seller");
        map.put("gmail", sharedPreference.getString(OtpVerify.this, "resend"));

        RetrofitAPI retrofitAPI = RetrofitApiClient.getRetrofit().create(RetrofitAPI.class);
        Call<ArrayList<SendOtppojo>> call = retrofitAPI.getotp(map);
        call.enqueue(new Callback<ArrayList<SendOtppojo>>() {
            @Override
            public void onResponse(Call<ArrayList<SendOtppojo>> call, Response<ArrayList<SendOtppojo>> response) {
                if (response.isSuccessful()) {

                    String result = new Gson().toJson(response.body());
                    System.out.println("======response result:" + result);
                    send_otp.addAll(response.body());
                    sharedPreference.putString(OtpVerify.this, "register_otp", "" + send_otp.get(0).getOtp());


                }
                System.out.println("======response :" + response);
            }

            @Override
            public void onFailure(Call<ArrayList<SendOtppojo>> call, Throwable t) {
                System.out.println("======response t:" + t);
            }
        });


    }
}

File: app/src/main/java/nithra/gift/suggestion/shop/birthday/marriage/otp/OtpVerify.kt
package nithra.gift.suggestion.shop.birthday.marriage.otp

import android.content.Intent
import android.os.Bundle
import android.os.CountDownTimer
import android.text.Editable
import android.text.TextWatcher
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import nithra.gift.suggestion.shop.birthday.marriage.product_shop.ShopAdd
import nithra.gift.suggestion.shop.birthday.marriage.R
import nithra.gift.suggestion.shop.birthday.marriage.retrofit.CheckOtp
import nithra.gift.suggestion.shop.birthday.marriage.retrofit.RetrofitAPI
import nithra.gift.suggestion.shop.birthday.marriage.retrofit.RetrofitApiClient
import nithra.gift.suggestion.shop.birthday.marriage.retrofit.SendOtppojo
import nithra.gift.suggestion.shop.birthday.marriage.product_shop.SellerProfileProductList
import nithra.gift.suggestion.shop.birthday.marriage.support.SharedPreference
import nithra.gift.suggestion.shop.birthday.marriage.support.Utils_Class
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class OtpVerify : AppCompatActivity() {
    private var enterotp: TextView? = null
    private var editEmail: TextView? = null
    var enterOtp: EditText? = null
    var otp1: EditText? = null
    var otp2: EditText? = null
    var otp3: EditText? = null
    var otp4: EditText? = null
    private var verify: String? = null
    private var editOtp: String? = null
    var sharedPreference = SharedPreference()
    var sendOtp: ArrayList<SendOtppojo>? = null
    private var edit: ImageView? = null
    var back: ImageView? = null
    var extra: Bundle? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.fragment_enter_o_t_p)
        val ttv = findViewById<TextView>(R.id.timer)
        sendOtp = ArrayList()
        back = findViewById(R.id.back)
        enterotp = findViewById(R.id.enterotp)
        editEmail = findViewById(R.id.edit_email)
        otp1 = findViewById(R.id.otp_1)
        otp2 = findViewById(R.id.otp_2)
        otp3 = findViewById(R.id.otp_3)
        otp4 = findViewById(R.id.otp_4)
        edit = findViewById(R.id.edit)
        enterOtp = findViewById(R.id.enter_otp)
        back!!.setOnClickListener { finish() }
        editEmail!!.text = "" + sharedPreference.getString(this@OtpVerify, "resend")
        object : CountDownTimer(120000, 1000) {
            override fun onTick(millisUntilFinished: Long) {
                ttv.text = "If you didn't receive a otp? " + millisUntilFinished / 1000
            }

            override fun onFinish() {
                ttv.text = "If you didn't receive a otp? Resend"
                sharedPreference.putString(this@OtpVerify, "register_otp", "" + 0)
            }
        }.start()
        ttv.setOnClickListener {
            if (Utils_Class.isNetworkAvailable(this@OtpVerify)) {
                otpGenerate()
            } else {
                Utils_Class.toast_normal(this@OtpVerify, "Please connect to your internet")
            }
            object : CountDownTimer(120000, 1000) {
                override fun onTick(millisUntilFinished: Long) {
                    ttv.text = "If you didn't receive a otp? " + millisUntilFinished / 1000
                }

                override fun onFinish() {
                    ttv.text = "If you didn't receive a otp? Resend"
                }
            }.start()
        }
        edit!!.setOnClickListener {
            finish()
        }
        otp1!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable) {
                if (s.isNotEmpty()) {
                    otp2!!.requestFocus()
                }
            }
        })
        otp2!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable) {
                if (s.isNotEmpty()) {
                    otp3!!.requestFocus()
                } else {
                    otp1!!.requestFocus()
                }
            }
        })
        otp3!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable) {
                if (s.isNotEmpty()) {
                    otp4!!.requestFocus()
                } else {
                    otp2!!.requestFocus()
                }
            }
        })
        otp4!!.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {}
            override fun afterTextChanged(s: Editable) {
                if (s.isNotEmpty()) {
                    otp4!!.requestFocus()
                } else {
                    otp3!!.requestFocus()
                }
            }
        })
        enterotp!!.setOnClickListener {
            editOtp =
                otp1!!.text.toString().trim { it <= ' ' } + otp2!!.text.toString()
                    .trim { it <= ' ' } + otp3!!.text.toString()
                    .trim { it <= ' ' } + otp4!!.text.toString().trim { it <= ' ' }
            if (otp1!!.text.toString() == "" && otp2!!.text
                    .toString() == "" && otp3!!.text.toString() == "" && otp4!!.text
                    .toString() == ""
            ) {
                Utils_Class.toast_center(this@OtpVerify, "Enter your otp")
            } else {
                if (Utils_Class.isNetworkAvailable(this@OtpVerify)) {
                    otpVerify()
                } else {
                    Utils_Class.toast_normal(this@OtpVerify, "Please connect to your internet")
                }
            }
        }
    }

    private fun otpVerify() {
        Utils_Class.mProgress(this@OtpVerify)!!.show()
        verify = editOtp!!.trim { it <= ' ' }
        val map = HashMap<String, String?>()
        map["action"] = "check_otp"
        map["user_id"] = sharedPreference.getString(this@OtpVerify, "user_id")
        map["otp"] = verify!!
        val retrofitAPI = RetrofitApiClient.retrofit!!.create(
            RetrofitAPI::class.java
        )
        val call = retrofitAPI.checkotp(map)
        call.enqueue(object : Callback<ArrayList<CheckOtp>> {
            override fun onResponse(
                call: Call<ArrayList<CheckOtp>>,
                response: Response<ArrayList<CheckOtp>>
            ) {
                if (response.isSuccessful) {
                    if (response.body()!![0].status == "Success") {
                        sharedPreference.putInt(this@OtpVerify, "yes", 1)
                        sharedPreference.putInt(this@OtpVerify, "profile", 1)
                        enterOtp!!.text.clear()
                        val user = sharedPreference.getString(this@OtpVerify, "user_status")
                        if (user == "exiting") {
                            sharedPreference.putInt(this@OtpVerify, "profile", 2)
                            finishAffinity()
                            val i = Intent(this@OtpVerify, SellerProfileProductList::class.java)
                            startActivity(i)
                        } else if (user == "new") {
                            finishAffinity()
                            val i = Intent(this@OtpVerify, ShopAdd::class.java)
                            startActivity(i)
                        }
                    }
                    if (response.body()!![0].status == "failure") {
                        Utils_Class.toast_center(this@OtpVerify, "Invalid otp")
                    }
                    Utils_Class.mProgress!!.dismiss()
                }
            }

            override fun onFailure(call: Call<ArrayList<CheckOtp>>, t: Throwable) {
            }
        })
    }

    private fun otpGenerate() {
        Utils_Class.mProgress(this@OtpVerify)!!.show()
        val map = HashMap<String, String?>()
        map["action"] = "check_seller"
        map["gmail"] = sharedPreference.getString(this@OtpVerify, "resend")
        val retrofitAPI = RetrofitApiClient.retrofit!!.create(
            RetrofitAPI::class.java
        )
        val call = retrofitAPI.getotp(map)
        call.enqueue(object : Callback<ArrayList<SendOtppojo>?> {
            override fun onResponse(
                call: Call<ArrayList<SendOtppojo>?>,
                response: Response<ArrayList<SendOtppojo>?>
            ) {
                if (response.isSuccessful) {
                    sendOtp!!.addAll(response.body()!!)
                    sharedPreference.putString(
                        this@OtpVerify,
                        "register_otp",
                        "" + sendOtp!![0].otp
                    )
                    Utils_Class.mProgress!!.dismiss()
                }
            }

            override fun onFailure(call: Call<ArrayList<SendOtppojo>?>, t: Throwable) {
            }
        })
    }
}